ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/Run.hs

```diff
@@ -201,24 +205,40 @@ startBlockFetchConnectionThread
 dispatchTick :: forall m blk.
   IOLike m =>
   Tracer m (TraceSchedulerEvent blk) ->
-  Tracer m () ->
-  ChainDB m blk ->
   StrictTVar m (Map PeerId (ChainSyncClientHandle m blk)) ->
   Map PeerId (PeerResources m blk) ->
+  NodeLifecycle blk m ->
+  LiveNode blk m ->
   (Int, (DiffTime, Peer (NodeState blk))) ->
-  m ()
-dispatchTick tracer stateTracer chainDb varHandles peers (number, (duration, Peer pid state)) =
+  m (LiveNode blk m)
+dispatchTick tracer varHandles peers lifecycle node (number, (duration, Peer pid state)) = do
   case peers Map.!? pid of
     Just PeerResources {prUpdateState} -> do
       traceNewTick
       atomically (prUpdateState state)
-      threadDelay duration
-      traceWith stateTracer ()
+      newNode <- checkDowntime
+      traceWith (lnStateTracer newNode) ()
+      pure newNode
     Nothing -> error "“The impossible happened,” as GHC would say."
   where
+    checkDowntime
+      | Just minInterval <- nlMinDuration
+      , duration > minInterval
+      = do
+        threadDelay 0.00001
```

Comment:
Could we add a comment explaining why this is useful/necessary?

Reply:
looks like it's not, removed it ✅